airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -42,35 +67,56 @@ def read_records(
             # On read requests we should always have the catalog available
             raise MissingSchemaError("Expected `json_schema` in the configured catalog but it is missing.")
         parser = self.get_parser(self.config.file_type)
-        for file in self.list_files_for_this_sync(stream_slice):
+        for file_description in stream_slice["files"]:
+            file = RemoteFile.from_file_partition(file_description)
```

Comment:
It feels very odd that we need to rebuild this thing. Why don't we keep the RemoteFile in the stream_slices? Is it because of the debug logs for the slice?

Reply:
I'm fine keeping them as RemoteFile in the slices. updated!